xds/internal/test/xds_integration_test.go

```diff
@@ -51,3 +66,148 @@ type testService struct {
 func (*testService) EmptyCall(context.Context, *testpb.Empty) (*testpb.Empty, error) {
 	return &testpb.Empty{}, nil
 }
+
+var (
+	// Globals corresponding to the single instance of the xDS management server
+	// which is spawned for all the tests in this package.
+	managementServer *e2e.ManagementServer
+	xdsClientNodeID  string
+)
+
+func init() {
+	// The management server is started and stopped from TestMain(), but the
+	// leakcheck runs after every individual test. So, we need to skip the
+	// goroutine which spawns the management server and is blocked on the call
+	// to `Serve()`.
+	leakcheck.RegisterIgnoreGoroutine("e2e.StartManagementServer")
```

Comment:
Put this in `TestMain` right where we do `setupManagementServer` instead of `init`?

Reply:
The documentation around `RegisterIgnoreGoroutine` said use it only from `init`. But yeah, `TestMain` should be fine too. Moved.

Reply:
Yeah, as long as it's done before leakcheck is ever called, then it's fine.